WallsAndHoles/tiletemplate.h

```diff
@@ -37,6 +48,10 @@ class TileTemplate : public QObject
 
     QColor color() const { return mColor; }
 
+
+    const TileMaterial &material() const { return *mMaterial; }
```

Comment:
Just return pointers? Not sure there is much value in passing around refs instead of pointers?